app/javascript/audio-recorder.js

```diff
@@ -0,0 +1,63 @@
+import { DirectUpload } from "activestorage"
+
+function responseFileName (codec) {
+  const fileTypes = {
+    'audio/wav': 'wav',
+    'audio/webm;codecs=opus': 'webm',
+    'audio/ogg;codecs=opus': 'ogg'
+  };
+
+  return `response-${Date.now()}.${fileTypes[codec]}`;
+}
+
+document.addEventListener("DOMContentLoaded", function(){
+  const input = document.querySelector('#story_audio')
```

Comment:
Since this executes on every page, `input === null` which will cause runtime errors. Can you do a Rails thing to prevent this script from loading/executing on other pages or use `window.location.pathname` in JS to isolate this whole file?

Reply:
Since it uses web pack I discovered:
```
if(window.location.pathname === '/stories/new') {
  import('../audio-recorder.js')
  .then()
}
```
seems to work.